### PR TITLE
Revise routing topic

### LINKED ISSUE #1876: Wrong parameters in $this->_forward() on extensions-dev-guide/routing.html
The page [https://devdocs.magento.com/guides/v2.2/extension-dev-guide/routing.html](https://devdocs.magento.com/guides/v2.2/extension-dev-guide/routing.html) (corresponding github page [here](https://github.com/magento/devdocs/blob/develop/guides/v2.1/extension-dev-guide/routing.md)) is wrong about the $this->_forward() function.

According to the page the function have one parameter containing the route, when the function really looks like this: _forward($action, $controller = null, $module = null, array $params = null). Source code [here](https://github.com/magento/magento2/blob/2.2-develop/lib/internal/Magento/Framework/App/Action/Action.php).

The page also seems to have other inaccurate statements, but I'm not familiar enough with Magento to know how they are wrong.

### PR DESCRIPTION
**Pull request by jcalcaben**
## This PR is a:
- [ ] New topic
- [X] Content fix or rewrite
- [ ] Bug fix or improvement
 

## Summary
 
When this pull request is merged, it will rewrite the Routing topic.
 

## Additional information

This PR fixes #1876

### CODE REVIEW

**Comment by jcalcaben** [comment]
whatsnew
Rewrite the [Routing](https://devdocs.magento.com/guides/v2.1/extension-dev-guide/routing.html) topic.

File: guides/v2.1/extension-dev-guide/routing.md [hunk 1]
```diff
@@ -1,107 +1,185 @@
...
-{% endhighlight %}
+A Magento {% glossarytooltip a05c59d3-77b9-47d0-92a1-2cbffe3f8622 %}URL{% endglossarytooltip %} that uses the standard router has the following format: 
+
+```
+<store-url>/<front-name>/<controller-name>/<action-name>
+```
+
+Where:
+
+* `<store-url>` - the base URL for the Magento instance
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/routing.md]: Add "specifies" for consistency.

File: guides/v2.1/extension-dev-guide/routing.md [hunk 2]
```diff
@@ -1,107 +1,185 @@
...
 
-<div class="bs-callout bs-callout-info" id="info">
-  <p><code>%routeId%</code> must be at least three characters in length and can consist of the following characters: <code>A-Z, a-z, 0-9, _</code>.</p>
-  <p><code>%frontName%</code> must be at least three characters in length and can consist of the following characters: <code>A-Z, a-z, 0-9, _, -</code>.</p>
-</div>
+### Default router
 
-To retrieve the configuration for route for an area by the specified router, use the `Magento\App\Framework\Route\Config`.
+The default router, defined by the [`DefaultRouter`] class, is the last router Magento checks during the routing process.
+Requests that reach this point often contain invalid URL that previous routers cannot handle.
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/routing.md]: plural: URLs

File: guides/v2.1/extension-dev-guide/routing.md [hunk 3]
```diff
@@ -1,107 +1,185 @@
...
 
-To replace the controller action in a route with custom one, add the custom controller class before the original controller.
+Magento uses the default [NoRouteHandler] to process these requests, but
+you can write your own no route handler by implementing the [NoRouteHandlerInterface].
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/routing.md]: no-route handler

File: guides/v2.1/extension-dev-guide/routing.md [hunk 4]
```diff
@@ -1,107 +1,185 @@
...
+Where:
 
-* The <a href="{{ site.mage2000url }}lib/internal/Magento/Framework/App/State.php" target="_blank">Magento\Framework\App\State</a>  class provides information on the state of the application, that is, current mode, installation date, and so on.
-* The <a href="{{ site.mage2000url }}lib/internal/Magento/Framework/App/AreaList.php" target="_blank">Magento\Framework\App\Arealist</a> class serves to configure the application areas through the `di.xml` file
-* The <a href="{{ site.mage2000url }}lib/internal/Magento/Framework/App/Area/FrontNameResolverInterface.php" target="_blank">Magento\App\Area\FrontNameResolverInterface</a> class resolves the dynamic area's front names
+* `%routerId` - The name of the router in Magento.    
+    See the reference tables in the [Router class section].
+* `%routeId%` - The unique node id for this route in Magento.
+* `%frontName%` - The first segment after the base URL of a request.
+* `%moduleName%` - The name of your module.
```
Comment (bdenham) [inline guides/v2.1/extension-dev-guide/routing.md]: Either make this list consistent with the other list above that uses a pattern of "specifies the..." or make the other list consistent with this one. :)

File: guides/v2.1/extension-dev-guide/routing.md [hunk 5]
```diff
@@ -1,107 +1,185 @@
...
-        <route id="%routeId%" frontName="%frontName%">
-            <module name="%moduleName%" before="%moduleName%"/>
-        </route>
-    </router>
-</config>
-{% endhighlight %}
+A Magento {% glossarytooltip a05c59d3-77b9-47d0-92a1-2cbffe3f8622 %}URL{% endglossarytooltip %} that uses the standard router has the following format: 
+
+```
+<store-url>/<front-name>/<controller-name>/<action-name>
```
Comment (eric-bohanon) [inline guides/v2.1/extension-dev-guide/routing.md]: I think we should also include the optional store code in between `<store-url>` and `<front-name>` (which determines store/website context)

**Pull request merged**